### DISCUSSION TITLE
Is anyone else interested in replacing the BambuLab dll for lan only mode and getting rid of all the delays and timeouts

### DISCUSSION THREAD

**Discussion by julie777** [OrcaSlicer/OrcaSlicer#810] [Ideas]
I use my P1P LAN only. 
Even though the printer is lan only Bambu/Orca slicer will not maintain the connection to the printer. Changing from the device page while connected and viewing status to the preview page and starting a print will bring up the print dialog which has to connect again and for me always times out. I find this terribly annoying. 

Looking in the network log file the slicer, even though it knows the printer is LAN only will always try to access the BL cloud. 

Sending the print file often results in failure to parse file, and requires sending again.

(In addition Orca doesn't switch to the device page after sending the file to the printer.)

I have already used MQTT to connect to my printer and monitor it. I haven't tracked the job send, or the camera feed.

Before I start trying to create a pull request for Orca, I want to know:
  - if there is a general desire to eliminate the proprietary bambu code when using LAN only mode.
  - if anyone else is willing to help reverse engineer the communication, which could possibly allow the X1 camera to work in LAN only mode.
  - if @SoftFever is willing to accept pull requests

**Comment by SoftFever** [comment]
@julie777 I love the idea!

> `if @SoftFever is willing to accept pull requests`

Absolutely! When you have an initial working copy ready, please let me know so I can create a separate branch for you. This will allow more people to try it out and provide feedback during development.

Also, I suggest adding an option for users to switch between Bambusource.dll and your implementation, providing them with more choice.

**Comment by julie777** [comment]
@SoftFever 
Can you point me to any references that would help me get up to speed on the codebase quickly?
I mostly would like the understand the structure and organization so I don't spend all my time searching for things.

For example off the top of my head:
  - where is the code that starts a print job?
  - where is the code for the device status page?
  - where is the main networking code and its use?

**Reply by dohn-joh** [reply]
> if there is a general desire to eliminate the proprietary bambu code when using LAN only mode.

I think quite a few people would be interested in this, but github discussions probably won't get the best visibility. 
I'm curious, has any progress been made with this?

**Reply by apalert** [reply]
Yep, very interested. Frustrating delays and timeouts happen a lot!

**Reply by SoftFever** [reply]
I would suggest start with this file: src/slic3r/Utils/NetworkAgent.hpp
It defines all the functions available in BambuSource.dll.
You will need to implement these functions to substitute BambuSource.dll.

**Comment by julie777** [comment]
@pujvz 

I have manually sent a file and printed it without using any BBL code. 
I am still trying to build Orca Slicer. I cannot seem to get any help, so I cannot make progress implementing the feature.

**Reply by SoftFever** [reply]
Please refer to #1425 for compiling helps.

**Reply by lin-ycv** [reply]
> I have manually sent a file and printed it without using any BBL code. I am still trying to build Orca Slicer. I cannot seem to get any help, so I cannot make progress implementing the feature.

any update? or care to share the code? maybe someone else can help compile and test?

**Comment by cybrsage** [comment]
I love the idea, especially with the ability to use a software switch to flip between the original and the modified dll.  I wish I could help, but my programming knowledge stopped with BASIC...many decades ago.

**Comment by WZ9V** [comment]
Would love this.

**Comment by rapapar** [comment]
Good idea, any update since ?

**Comment by JMcrafter26** [comment]
Are there any updates on this? (I am asking because of the new BBL firmware update and Bambu connection)

**Reply by dmorris68** [reply]
Hopefully the newly announced Developer Mode is the path forward here, otherwise the new BBL direction might have killed this approach entirely since it wouldn't allow unauthorized access to send anything to the printer. At least not right through the front door. :)
